tools/driver-cxx/driver-cxx.cpp

```diff
@@ -155,24 +159,29 @@ int main(int argc, char **argv) {
 
   mull::TestFramework testFramework(testFrameworkOption.testFramework(toolchain, configuration));
 
-  bool junkDetectionEnabled = false;
+  bool bitcodeCompilationDatabaseAvailable = false;
+  bool compilationDatabasePathAvailable = false;
+  bool bitcodeCompilationFlagsAvailable = false;
   std::string cxxCompilationFlags;
   std::string cxxCompilationDatabasePath;
   if (!bitcodeCompilationFlags.empty()) {
     diagnostics.info(std::string("Found compilation flags in the input bitcode"));
-    junkDetectionEnabled = true;
+    bitcodeCompilationDatabaseAvailable = true;
   }
   if (!tool::CompilationFlags.empty()) {
     cxxCompilationFlags = tool::CompilationFlags.getValue();
-    junkDetectionEnabled = true;
+    bitcodeCompilationFlagsAvailable = true;
   }
   if (!tool::CompilationDatabasePath.empty()) {
     cxxCompilationDatabasePath = tool::CompilationDatabasePath.getValue();
-    junkDetectionEnabled = true;
+    compilationDatabasePathAvailable = true;
   }
+  bool compilationDatabaseInfoAvailable = bitcodeCompilationDatabaseAvailable ||
+                                          compilationDatabasePathAvailable ||
+                                          bitcodeCompilationFlagsAvailable;
```

Comment:
Could you explain the need for these three extra variables? It feels a bit too noisy and adds more mental overhead than needed.

Reply:
This is the only comment where I want to make an argument for actually doing this.

We can make the variables a little bit shorter but they actually make making of decisions below very clear:

We only enter the "Junk detection" decision only if `compilationDatabaseInfoAvailable` and we enter the "White search" decision if `EnableASTSearch && compilationDatabaseInfoAvailable`.

Otherwise I would have to introduce a variable called `whiteSearchIsEnabled` and put it where `junkDetectionEnabled` is set to `true` and we would have a duplication of both variables set to true to make their decisions.

This makes sense to me. For you?

Reply:
It does make sense to me, though there is a small misunderstanding on either side.
What I meant is to basically rename `junkDetectionEnabled` into `compilationDatabaseInfoAvailable` and drop the unnecessary variables. Then the same logic would be preserved, right?

I'm not going to die on this hill, so if having three extra variables works better for you, then I don't have any objections :)